bikeshed/__init__.py

```diff
@@ -746,14 +746,28 @@ def processAutolinks(doc):
         if linkType in ("property", "descriptor", "propdesc") and "*" in linkText:
             continue
 
-        url = doc.refs.getRef(linkType, linkText,
+        ref = doc.refs.getRef(linkType, linkText,
                               spec=el.get('data-link-spec'),
                               status=el.get('data-link-status'),
                               linkFor=el.get('data-link-for'),
                               el=el,
                               error=(linkText.lower() not in doc.md.ignoredTerms))
-        if url is not None:
-            el.set('href', url)
+        # Capture the reference (and ensure we add a biblio entry) if it
+        # points to an external specification. We check the spec name here
+        # rather than checking `status == "local"`, as "local" refs include
+        # those defined in `<pre class="anchor">` datablocks, which we do
+        # want to capture here.
+        if ref and ref.spec is not None and ref.spec is not "" and ref.spec != doc.refs.specVName:
+            if ref.spec not in doc.externalRefsUsed:
```

Comment:
Use defaultdict().  These kinds of checks are verbose, error-prone, and avoidable.